nexus_constructor/instrument_view.py

```diff
@@ -218,40 +195,29 @@ def add_component(
 
         if positions is None:
             positions = [QVector3D(0, 0, 0)]
-        mesh = OffMesh(geometry.off_geometry)
-
-        self.component_meshes[name] = mesh
-        position_transforms = []
-        for position in positions:
-            transform = Qt3DCore.QTransform()
-            transform.setTranslation(position)
-            position_transforms.append(transform)
-        self.component_positions[name] = position_transforms
-        # Note, the a list comprehension like below doesn't work (end up with segfault)
-        # self.component_positions[name] = [Qt3DCore.QTransform().setTranslation(position) for position in positions]
-        self._create_entities(name)
-
-    def _create_entities(self, name):
+        mesh = OffMesh(geometry.off_geometry, self.component_root_entity)
+
         with DetachedRootEntity(
             self.component_root_entity, self.combined_component_axes_entity
         ):
-            self.component_entities[name] = [
-                create_qentity(
-                    [
-                        self.component_meshes[name],
-                        self.grey_material,
-                        position_transform,
-                    ],
-                    self.component_root_entity,
+            for position in positions:
+                transform = Qt3DCore.QTransform(self.component_root_entity)
+                transform.setTranslation(position)
+
+                material = create_material(
+                    QColor("black"), QColor("grey"), self.component_root_entity
                 )
-                for position_transform in self.component_positions[name]
-            ]
+
+                self.component_entities[name] = [
```

Comment:
This isn't quite right, now you overwrite `self.component_entities[name]` with a list contain a single entity each time.
The result is that we only have a handle for the last pixel that was created, so when we try to delete a detector bank component only that pixel is removed from the 3D view.

Reply:
Good spot, thanks.